File: DnDManagementSystem/Front-End/DnDManager/src/app/models/temp-models/Item.ts
export class Item{
  constructor(public _itemId:number, public _name:string, public _weight:string, public _value:string, public _description:string, public _inventorySheet:number) {}

  get itemId(): number {
    return this._itemId;
  }

  set itemId(value: number) {
    this._itemId = value;
  }

  get inventorySheet(): number {
    return this._inventorySheet;
  }

  set inventorySheet(value: number) {
    this._inventorySheet = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get weight(): string {
    return this._weight;
  }

  set weight(value: string) {
    this._weight = value;
  }

  get value(): string {
    return this._value;
  }

  set value(value: string) {
    this._value = value;
  }

  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }
}

File: DnDManagementSystem/Back-End/DnD_SpringBoot_v2/src/main/java/com/revature/models/charactersheet/Item.java
package com.revature.models.charactersheet;

import java.util.ArrayList;
import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity

//Lombok
@Getter
@Setter
@NoArgsConstructor
//@AllArgsConstructor
@ToString
@EqualsAndHashCode
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "itemId")
@JsonIgnoreProperties(value = {"hibernateLazyInitializer", "handler"})
@Table(name = "item_db")

public class Item {
	@Id
	@Column(name = "itemId") 
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	int itemId;
	
	@Column(name = "name")
	String name;
	@Column(name = "weight")
	String weight;
	@Column(name = "value")
	String value;
	@Column(name = "description")
	String description;
	
	
	@ManyToOne
	@JoinColumn(name = "inventorySheetId")
	private InventorySheet inventorySheet;
	
	
	
}
